### ISSUE TITLE
LC_Invite is never defined

### ISSUE THREAD

**Issue by FraSis** [livechat/angular-livechat#12]
Hi,

I am using the angular-livechat component on an angular 5 application.

Tests made with the 0.0.3 and 0.0.6 version of this component.

When I call the `LC_API.reload_window()` method I got this exception :

`ERROR ReferenceError: "LC_Invite is not defined"
send https://cdn.livechatinc.com/tracking.js:7:9015
reload_window https://cdn.livechatinc.com/tracking.js:7:28204`

What I want to do is to reload livechat when a user log in or out the application to set the proper `__lc.visitor` value.
I want to see "Client" in the agent chats windows for an unauthenticated user and the first name and last name of an authenticated user.
Maybe I am doing something wrong.

**Comment by ZiggyKraus** [comment]
Hello @FraSis !

Sorry for the late response regarding this issue. I'm not an Angular developer myself, but I will do my best to help you with this particular case :)

@FraSis , the `LC_API.reload_window()` throws the:

`ERROR ReferenceError: "LC_Invite is not defined" send https://cdn.livechatinc.com/tracking.js:7:9015 reload_window https://cdn.livechatinc.com/tracking.js:7:28204`

message because the `LC_API.reaload_window()` the method is not supported by our latest chat widget (I do not have the ID of your LiveChat license, but since you are getting the `LC_Invite is not defined` error message, it is safe to assume that you are using the **latest chat widget**).

Also, I'm afraid that there's no good way to set up the `lc.visitor` object after your LiveChat app has already loaded, as in order to populate `lc.visitor` with new data, you will have to make sure that the LiveChat code itself has been reloaded (and not the chat widget alone, with the use of methods like `LC_API.reload_window()`). The thing that I can recommend is, for example, applying some sort of **slow load** mechanism within your Angular app. What do I mean by that?

Basically you could prevent your LiveChat from loading until your application's user authenticates himself (thus making the `name` and `email` fields available) and until the user will manually click on a button like **Chat with us**: only after those two conditions will be met, your app could load LiveChat code with `lc.visitor` object added to it.

@FraSis I hope that the suggested workaround will allow you to achieve your goal with LiveChat for Angular. Once again, so sorry for such a late reply and for not having the reload method that would update the `lc.visitor` object in real time: this is definitely something that we will add to our feature request list, but I cannot promise when it would be applied to the chat widget itself 😞

**Comment by ZiggyKraus** [comment]
@FraSis a quick update here:

After posting my initial reply I have thought about it once again and I think that you could try using the `LC_API.set_visitor_name('John Doe');` and `LC_API.set_visitor_email('john@doe.com');` methods that can update the `lc.visitor` object in real time. Both of those methods are described here: 
https://developers.livechatinc.com/docs/js-api/#set-visitor-name

and here:
https://developers.livechatinc.com/docs/js-api/#set-visitor-email

so feel free to check them out!

So sorry for no mentioning them earlier: those methods were implemented only recently, that's why I have not thought about them in a first place

**Comment by FraSis** [comment]
@ZiggyKraus Thank you for your responses, I will try those options asap and let you know if it is solving my issue.

**Comment by ZiggyKraus** [comment]
@FraSis you're more than welcome! Always happy to help 🙂

**Comment by rmcsharry** [comment]
@ZiggyKraus Are you able to please update the readme to explain, with maybe example code, how to access the LC_API.set_visitor_name from within a typescript component? Thanks.

Or maybe update the stackblitz?

**Comment by dvelazdiaz** [comment]
Hello @ZiggyKraus, it would awesome if you can show us how access to the LC_API from an angular component

**Comment by MichalPaszowski** [comment]
Hi,

We apologise that you had to wait so long for a response.
We have prepared a new version of our library. This repository has been updated and the library can be found on NPM https://www.npmjs.com/package/@livechat/angular-widget.

Please give it a try and if necessary open new issue.
